test/serializer.jl

```diff
@@ -0,0 +1,75 @@
+module TestSerializer
+
+using JSON
+using Base.Test
+using Compat
+
+# to define a new serialization behaviour, import these first
+import JSON.Serializations: CommonSerialization, StandardSerialization
+import JSON: StructuralContext
+
+# those names are long so we can define some type aliases
+typealias CS CommonSerialization
+typealias SC StructuralContext
+
+# for test harness purposes
+function sprint_kwarg(f, args...; kwargs...)
+    b = IOBuffer()
+    f(b, args...; kwargs...)
+    takebuf_string(b)
+end
+
+# issue #168: Print NaN and Inf as Julia would
+immutable NaNSerialization <: CS end
+JSON.show_json(io::SC, ::NaNSerialization, f::AbstractFloat) =
+    Base.print(io, f)
```

Comment:
Very nice, It's hard to imagine custom support for NaN and Inf being simpler than this :-)